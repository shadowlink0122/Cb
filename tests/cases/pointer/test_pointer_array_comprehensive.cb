// ポインタの配列の包括的テスト

void main() {
    // テスト1: 基本的な宣言と初期化
    println("=== Test 1: Basic pointer array ===");
    int a = 10, b = 20, c = 30;
    int*[3] ptrs = [&a, &b, &c];
    
    println(*ptrs[0]);  // 10
    println(*ptrs[1]);  // 20
    println(*ptrs[2]);  // 30
    
    // テスト2: 個別の変数へのポインタ
    println("=== Test 2: Individual variables ===");
    int v1 = 1, v2 = 2, v3 = 3, v4 = 4, v5 = 5;
    int*[5] arr_ptrs = [&v1, &v2, &v3, &v4, &v5];
    
    for (int i = 0; i < 5; i++) {
        println(*arr_ptrs[i]);  // 1, 2, 3, 4, 5
    }
    
    // テスト3: ポインタ経由での値の変更
    println("=== Test 3: Modify through pointers ===");
    int val1 = 111, val2 = 222;
    int*[2] value_ptrs = [&val1, &val2];
    *value_ptrs[0] = 100;
    *value_ptrs[1] = 500;
    println(val1);  // 100
    println(val2);  // 500
    
    // テスト4: 複数のポインタ配列
    println("=== Test 4: Multiple pointer arrays ===");
    int x = 1, y = 2, z = 3;
    int*[3] set1 = [&x, &y, &z];
    int*[3] set2 = [&a, &b, &c];
    
    println(*set1[0]);  // 1
    println(*set1[1]);  // 2
    println(*set1[2]);  // 3
    println(*set2[0]);  // 10
    println(*set2[1]);  // 20
    println(*set2[2]);  // 30
    
    println("=== All pointer array tests passed ===");
}
