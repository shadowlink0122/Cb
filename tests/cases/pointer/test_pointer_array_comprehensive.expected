=== Test 1: Basic pointer array ===
10
20
30
=== Test 2: Individual variables ===
1
2
3
4
5
=== Test 3: Modify through pointers ===
100
500
=== Test 4: Multiple pointer arrays ===
1
2
3
10
20
30
=== All pointer array tests passed ===
